### PR TITLE
Mode  of Generator::ircMode() and GeneratorInterface::ircMode() optional

### PR DESCRIPTION
**Pull request by rpasing**

### CODE REVIEW

File: src/Generator.php [hunk 1]
```diff
@@ -180,7 +180,7 @@ public function ircPart($channels, $message = null)
      * @param string $param
      * @return string
      */
-    public function ircMode($target, $mode, $param = null)
+    public function ircMode($target, $mode = null, $param = null)
```
Comment (elazar) [inline src/Generator.php]: Type for `$mode` in the docblock needs to be updated to `string|null`.

File: src/GeneratorInterface.php [hunk 2]
```diff
@@ -130,7 +130,7 @@ public function ircPart($channels, $message = null);
      * @return string
      * @link https://tools.ietf.org/html/rfc2812#section-4.2.3
      */
-    public function ircMode($target, $mode, $param = null);
+    public function ircMode($target, $mode = null, $param = null);
```
Comment (elazar) [inline src/GeneratorInterface.php]: Type for `$mode` in the docblock needs to be updated to `string|null`.

**Comment by elazar** [comment]
This also needs a test case. Can you add one around [here](https://github.com/phergie/phergie-irc-generator/blob/6df0832ea8b59eb12a4fe14d59698d0c4aa2e198/tests/GeneratorTest.php#L100)?

**Comment by rpasing** [comment]
@elazar PHPDocs updated, testcase added!

**Comment by elazar** [comment]
@rpasing Tests fail because the expected command to be generated needs to have its trailing parameter prefixed by a colon (`:`).

**Comment by rpasing** [comment]
@elazar Argghhhh.... Sorry! Fixed

**Comment by elazar** [comment]
Thanks!

**Pull request merged**